File: app/src/main/java/com/example/giniappproject/clock/ClockApplication.java
package com.example.giniappproject.clock;

import android.app.Application;

import io.realm.Realm;
import io.realm.RealmConfiguration;

public class ClockApplication extends Application {


    @Override
    public void onCreate() {
        super.onCreate();

        Realm.init(this);

        RealmConfiguration realmConfiguration = new RealmConfiguration.Builder()
                .name("clock_project.realm")
                .encryptionKey(new byte[64])
                .schemaVersion(1)
                .build();


        Realm.setDefaultConfiguration(realmConfiguration);

    }
}

File: app/src/main/java/com/example/clockproject/clock_app/ClockApplication.kt
package com.example.clockproject.clock_app

import android.app.Application
import org.koin.android.ext.koin.androidContext
import org.koin.core.context.startKoin

class ClockApplication : Application() {
    override fun onCreate() {
        super.onCreate()

        startKoin {
            // Android context
            androidContext(applicationContext)
            // modules
            modules(ClockModules().createModules(applicationContext))
        }
    }
}